torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -990,10 +994,14 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // This is the signal from watchdog threads to indicate whether the monitor
   // thread should dump. Making it static so that it is accessiable from all the
   // PGs. With this flag, monitor thread would dump debug info under any one of
-  // the 3 conditions: 1: this flag is set to true by the watchdog thread when
-  // it detects a timeout. 2: timeout signal is received from
-  // other ranks through tcpstore 3: no heartbeat of watchdog Note that only the
-  // monitor thread from PG0 should dump the debug info and only once
+  // the three conditions:
+  //
+  // 1: watchdog thread of any PG detects a collective timeout.
+  // 2: timeout signal is received from other ranks through tcpstore.
+  // 3: no heartbeat detected of the watchdog of the current PG.
```

Comment:
current PG's watchdog heartbeat timeout occurs

Reply:
I changed "heartbeat timeout" to "heartbeat missing"